pkg/kconfig/kconfig.go

```diff
@@ -231,3 +242,61 @@ func IsInContainer() (bool, error) {
 
 	return false, nil
 }
+
+// CheckStackUnwindingEnabled checks whether stack unwinding is enabled or not.
+func CheckStackUnwindingEnabled() error {
+	configPaths, err := getConfigPaths()
+	if err != nil {
+		return err
+	}
+
+	var result *multierror.Error
+	for _, configPath := range configPaths {
+		if _, err := os.Stat(configPath); err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			result = multierror.Append(result, err)
+			continue
+		}
+		kernelConfig, err := getConfig(configPath)
+		if err != nil {
+			return err
+		}
+		stackUnwindingEnabled, err := checkConfigOption(kernelConfig, stackUnwinderConfig)
+		if err != nil {
+			return err
+		}
+		// The only success case.
+		if stackUnwindingEnabled {
+			return nil
+		}
+	}
+
+	if result != nil && len(result.Errors) > 0 {
+		return result.ErrorOrNil()
+	}
+
+	// If we reach this point, either we have not found a config file or required options are not enabled.
+	return fmt.Errorf("kernel config not found, tried paths: %s", strings.Join(configPaths, ", "))
+}
+
+// CheckCgroupVersion checks whether the system supports either cgroup v1 or v2
+// https://kubernetes.io/docs/concepts/architecture/cgroups/#check-cgroup-version
+func CheckCgroupVersion() error {
+	cgroupV1Output := "tmpfs"
+	cgroupV2Output := "cgroup2fs"
+
+	cmd := exec.Command("stat", "-fc", "%T", "/sys/fs/cgroup")
```

Comment:
Perhaps we could use [`os.Stat`](https://pkg.go.dev/os#File.Stat) instead of relying on `stat(1)`